tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -113,44 +119,81 @@ private async void GetIcons()
         }
     }
 
-    internal string CreateFullCommandLine(int? pid, HWND? hwnd)
-    {
-        return "\"" + Executable + "\"" + CreateCommandLine(pid, hwnd);
-    }
-
-    internal string CreateCommandLine(int? pid, HWND? hwnd)
-    {
-        var commandLine = $" {OtherArgs}";
-
-        if (ArgType == ExternalToolArgType.Hwnd && hwnd is not null)
-        {
-            commandLine = $" {ArgPrefix} {hwnd:D} {OtherArgs}";
-        }
-        else if (ArgType == ExternalToolArgType.ProcessId && pid is not null)
-        {
-            commandLine = $" {ArgPrefix} {pid:D} {OtherArgs}";
-        }
-
-        return commandLine;
-    }
+    internal async Task<bool> Invoke(int? pid, HWND? hwnd)
```

Comment:
Invoke is ambiguous.  What is doing the invoking, and what is the purpose of being invoked?  Looks like this is used to launch/active an app?

Reply:
PI is invoking this external tool.